src/deluge/storage/storage_manager.cpp

```diff
@@ -1026,10 +1026,14 @@ int32_t StorageManager::createFile(FIL* file, char const* filePath, bool mayOver
 	return NO_ERROR;
 }
 
-int32_t StorageManager::createXMLFile(char const* filePath, bool mayOverwrite) {
+int32_t StorageManager::createXMLFile(char const* filePath, bool mayOverwrite, bool displayErrors) {
 
 	int32_t error = createFile(&fileSystemStuff.currentFile, filePath, mayOverwrite);
 	if (error) {
+		if (displayErrors) {
```

Comment:
nit: can you just && these instead of nesting?

Reply:
Actually it seems I can't, because if the error should not be displayed, the function still needs to return the error...